File: Leetcode/python/539-min-time-diff.py
from typing import List


class Solution:
    def findMinDifference(self, timePoints: List[str]) -> int:
        timePoints.sort()

        def timeToMin(t):
            h, m = map(int, t.split(":"))
            return 60 * h + m

        res = (
            24 * 60 -
            timeToMin(timePoints[-1]) +
            timeToMin(timePoints[0])
        )
        for i in range(len(timePoints) - 1):
            curr = timeToMin(timePoints[i + 1])
            prev = timeToMin(timePoints[i])
            diff = curr - prev
            res = min(res, diff)

        return res

File: Leetcode/typescript/539-min-time-diff.ts
function findMinDifference(timePoints: string[]): number {
  timePoints.sort();

  function timeToMin(t: string) {
    const [h, m] = t.split(":").map(Number);
    return 60 * h + m;
  }

  let res =
    24 * 60 -
    timeToMin(timePoints[timePoints.length - 1]) +
    timeToMin(timePoints[0]);
  for (let i = 0; i < timePoints.length - 1; i++) {
    const curr = timeToMin(timePoints[i + 1]);
    const prev = timeToMin(timePoints[i]);
    const diff = curr - prev;
    res = Math.min(res, diff);
  }
  return res;
}
